README.md

```diff
@@ -44,3 +44,164 @@ information.
 
 If you pass `--dev`, the `devDependencies` will be checked as well as the
 `dependencies`.
+
+`jsgl` also checks sub-packages for `--local` and `--pr` flags when it
+detects that the repository is a monorepo. It assumes a certain directory
+structure for detecting whether a repository is a monorepo: the top-level
+directory should have the `packages` directory in it and sub-packages must
+exist under that directory. In that case, all the `package.json` files are
+found from sub-packages and `jsgl` checks all of them.
+
+For example, when a directory `foo` is like this:
+```
+foo
+ |
+ +-- packages
+ |    |
+ |    +-- bar
+ |    |    |
+ |    |    +-- package.json
+ |    |    |
+ |    |    +-- ...
+ |    |
+ |    +-- baz
+ |         |
+ |         +-- package.json
+ |         |
+ |         +-- ...
+ |
+ +-- package.json
+ |
+ +-- ...
+```
+, `jsgl` checks all of `foo/package.json`, `foo/packages/bar/package.json`,
+and `foo/packages/baz/package.json`.
+
+## Configurations
+
+You can customize how `jsgl` works with the configuration file, named
+`js-green-licenses.json`. For example, you can specify the license list that
+you would like to consider green. The license IDs must be listed in the
+`greenLicenses` section of the configuration file. In that case, `jsgl` will
+use that custom list instead of its default list.
+
+You can also whitelist some NPM packages and they will be considered "green"
+even when they have non-green licenses or no licenses. It's useful when
+`jsgl` is unable to verify the validness of a certain package's license for
+some reason. For example, when a package doesn't specify its license in its
+`package.json` but has a separate `LICENSE` file, `jsgl` can't verify that.
+You can whitelist that package to make `jsgl` not complain about that
+package.
+
+A typical configuration file looks like this:
+```json
+{
+  "greenLicenses": [
+    "Apache-2.0",
+    "MIT",
+    "BSD-3-Clause",
+    ...
+  ],
+  "packageWhitelist": [
+    "foo",
+    "bar",
+    "package-with-no-license",
+    "package-with-okish-license",
+    ...
+  ]
+}
+```
+
+The `greenLicenses` section is for the custom license list and the
+`packageWhitelist` section is for the package whitelist.
+
+The configuration file must be located in the top-level directory of a
+repository for `--local` and `--pr`. When checking remote NPM packages,
+`jsgl` tries to find the location file in the current local directory from
+which `jsgl` is invoked.
+
+It is desirable that the license names in the `greenLicenses` section be
+valid license IDs defined in https://spdx.org/licenses/ whenever possible.
+
+## Interface as a Library
+
+You can also use `js-green-licenses` as a library as well as a command-line
+utility. Usually the `LicenseChecker` class is the only one you would have to
+use.
+
+### Instantiation
+
+```typescript
+const opts: LicenseCheckerOptions = {
```

Comment:
Maybe add a hint about omitting types so that JavaScript users (who aren't familiar with type syntax) can also use this. It might even be preferable to give JS examples.

Reply:
Changed to a JS example. TypeScript users will be able to figure out how to do things in TS from JS examples, I think.

Reply:
I left the examples in the "Interface as a Library" section as TS because I think it shows the type information of arguments and return types better.